samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java

```diff
@@ -354,15 +358,29 @@ public void stop() {
     LOG.info("Stopping event hub system consumer...");
     List<CompletableFuture<Void>> futures = new ArrayList<>();
     streamPartitionReceivers.values().forEach((receiver) -> futures.add(receiver.close()));
-    CompletableFuture<Void> future = CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()]));
+    CompletableFuture<Void> allFuture = CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()]));
     try {
-      future.get(DEFAULT_SHUTDOWN_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
-    } catch (ExecutionException | InterruptedException | TimeoutException e) {
+      allFuture.get(DEFAULT_SHUTDOWN_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+    } catch (Exception e) {
       LOG.warn("Failed to close receivers", e);
+      Util.logThreadDump("Log thread dump as we failed during consumer shutdown");
     }
+    LOG.info("Receivers are shutdown. Start shutting down EventHubManagers");
+    // There could be potentially many EventHubManagers, so close the managers in parallel
+    ExecutorService shutdownExecutorService = Executors.newCachedThreadPool();
+    List<Future> futureList = new ArrayList<>();
     perPartitionEventHubManagers.values()
-        .parallelStream()
-        .forEach(ehClientManager -> ehClientManager.close(DEFAULT_SHUTDOWN_TIMEOUT_MILLIS));
+        .forEach(ehClientManager -> futureList.add(
+            shutdownExecutorService.submit(() -> ehClientManager.close(DEFAULT_SHUTDOWN_TIMEOUT_MILLIS))));
+    futureList.forEach(future -> {
```

Comment:
Would it be possible to use  the same `CompletableFuture.allOf(DEFAULT_SHUTDOWN_TIMEOUT)` to aggregate the futures here?

Reply:
So the ehClientManager.close is already performing the timeout logic. That's sequently call "future.get()" shouldn't cause any issues. 
Plus, CompletableFuture.allOf only works for CompletableFuture<T> type